Dockerfile

```diff
@@ -0,0 +1,40 @@
+# Stage 1: Build the NestJS application
+FROM node:22-alpine AS builder
+
+WORKDIR /app
+
+# Copy package.json and package-lock.json for efficient layer caching
+COPY package*.json ./
+
+# Install dependencies
+RUN npm install
+
+# Copy the rest of the application code
+COPY . .
+
+# Build the application
+RUN npm run build
+
+# Stage 2: Production image
+FROM node:22-alpine AS production
+
+WORKDIR /app
+
+RUN apk add --no-cache gettext
+
+#Copy the compiled application from the build stage
+COPY --from=builder /app/dist ./dist
+COPY --from=builder /app/migrations ./migrations
+
+#Copy package.json and install only production dependencies
+COPY package.json ./
```

Comment:
why are we not copying the lock file here?

Reply:
It is not copied because it is not needed to run the application.

Reply:
yeah but it is required to download the dependencies right